src/coreclr/nativeaot/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/ExecutionEnvironmentImplementation.Runtime.cs

```diff
@@ -21,11 +21,6 @@ namespace Internal.Reflection.Execution
     //==========================================================================================================
     internal sealed partial class ExecutionEnvironmentImplementation : ExecutionEnvironment
     {
-        public sealed override object NewObject(RuntimeTypeHandle typeHandle)
-        {
-            return RuntimeAugments.NewObject(typeHandle);
```

Comment:
`RuntimeAugments.NewObject` checks for byref-like types. Where does the check for byref-like types happen now?

Also, we may be able to get rid of the arg validation from ``RuntimeAugments.NewObject`` now that it is not needed by this path.